### ISSUE TITLE
Bench methodology may not be representative to the setup of most users.

### ISSUE THREAD

**Issue by ejgallego** [rocq-prover/rocq#16616]
As far as I can tell, we have been using `bench` to measure potential improvements to Coq.

However, I think the test methodology may be a bit flawed in that most users caring about performance won't be running the compilation setup the bench does, but one with `flambda` (which often provides close to 10% improvement).

Thus, it seems to me that the current methodology for performance PR is not representative of real-world impact.

**Comment by Janno** [comment]
I think @rlepigre and I are running into this issue with #17839. We've seen local improvements that are entirely unreflected in the bench and the current state is much slower in the bench than it is locally on our machine. The kernel code we are tuning is extremely sensitive to inlining so the lack of flambda in the bench could explain why the overhead of the change reported by the bench is roughly twice what it is locally on my machine. (And that would mean that a decision to merge/not merge the MR based on the benchmark data might not be a good idea.)

**Comment by ejgallego** [comment]
Yes, I was afraid that this could happen; we should mitigate this bug.

Let's add a commit to run the bench with flambda.

**Comment by SkySkimmer** [comment]
I'm fine switching to flambda by default, anyone want to make a PR?

**Comment by SkySkimmer** [comment]
I tried switching to flambda recently but couldn't remember how it works, the code in bench.sh doesn't work (at least for 4.14.1).

**Comment by ejgallego** [comment]
Yes the code seems to have some strange stuff, what it is that you don't remember how it works? Installing flambda?

See the Docker file.
